apps/chat/src/components/Chat/ModelList.tsx

```diff
@@ -0,0 +1,168 @@
+import { IconChevronDown } from '@tabler/icons-react';
+import { useMemo, useState } from 'react';
+
+import classNames from 'classnames';
+
+import { groupModelsAndSaveOrder } from '@/src/utils/app/conversation';
+import { hasParentWithAttribute } from '@/src/utils/app/modals';
+
+import { OpenAIEntity } from '@/src/types/openai';
+
+import { useAppSelector } from '@/src/store/hooks';
+import { ModelsSelectors } from '@/src/store/models/models.reducers';
+
+import { ModelIcon } from '../Chatbar/ModelIcon';
+import { EntityMarkdownDescription } from '../Common/MarkdownDescription';
+import { ModelVersionSelect } from './ModelVersionSelect';
+
+interface ModelGroupProps {
+  entities: OpenAIEntity[];
+  selectedModelId: string | undefined;
+  onSelect: (id: string) => void;
+  notAllowExpandDescription?: boolean;
+}
+
+const ModelGroup = ({
+  entities,
+  selectedModelId,
+  onSelect,
+  notAllowExpandDescription,
+}: ModelGroupProps) => {
+  const [isOpened, setIsOpened] = useState(false);
+  const recentModelsIds = useAppSelector(ModelsSelectors.selectRecentModelsIds);
+
+  const currentEntity = useMemo(() => {
+    // if only 1 model without group
+    if (entities.length < 2) {
+      return entities[0];
+    }
+    // selected
+    const selected = entities.find((e) => e.id === selectedModelId);
+    if (selected) return selected;
+    // find latest used version
+    const minIndex = Math.min(
+      ...recentModelsIds
+        .map((rid) => entities.findIndex((e) => e.id === rid))
+        .filter((ind) => ind >= 0),
```

Comment:
nit: maybe it's better to set find !== -1? I mean it's more intuitive to expect -1 or something else, than >=0 or something else
I know this is discussible, but it's nit, so do whatever you want in current case :)